packages/x-charts/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,360 @@
+          rep.push({
+            type: seriesType,
+            id: seriesId,
+            // @ts-ignore
+            color: series[seriesType].series[seriesId].color,
+          });
+        }
+      });
+    });
+    return rep;
+  }, [USED_X_AXIS_ID, series]);
+
+  if (dataIndex == null) {
+    return null;
+  }
+
+  const markerSize = 30; // TODO: allows customization
+  const shape = 'square';
+  return (
+    <React.Fragment>
+      {axisValue != null && (
+        <React.Fragment>
+          <Typography variant="caption" sx={{ display: 'flex', alignItems: 'center' }}>
+            {xAxisName}: {axisValue.toLocaleString()}
+          </Typography>
+          <Divider />
+        </React.Fragment>
+      )}
+      {seriesConcerned.map(({ type, color, id }) => (
+        <Typography variant="caption" key={id} sx={{ display: 'flex', alignItems: 'center' }}>
+          <svg width={markerSize} height={markerSize}>
+            <path
+              // @ts-ignore TODO: Fix me
+              d={d3Symbol(d3SymbolsFill[getSymbol(shape)], markerSize)()!}
+              // TODO: Should be customizable. Maybe owner state would make more sense
+              // fill={invertMarkers ? d.stroke : d.fill}
+              // stroke={invertMarkers ? d.fill : d.stroke}
+              fill={color}
+              transform={`translate(${markerSize / 2}, ${markerSize / 2})`}
+            />
+          </svg>
+          {/* @ts-ignore */}
+          {id}: {format(series[type].series[id].data[dataIndex])}
+        </Typography>
+      ))}
+    </React.Fragment>
+  );
+}
+
+const useMouseTraker = () => {
+  const svgRef = React.useContext(SVGContext);
+
+  // Use a ref to avoid rerendering on every mousemove event.
+  const [mousePosition, setMousePosition] = React.useState<null | { x: number; y: number }>(null);
+
+  const handleMouseOut = React.useCallback(() => {
+    setMousePosition(null);
+  }, [setMousePosition]);
+
+  const handleMouseMove = React.useCallback(
+    (event: MouseEvent) => {
+      setMousePosition({
+        x: event.clientX,
+        y: event.clientY,
+      });
+    },
+    [setMousePosition],
+  );
+
+  React.useEffect(() => {
+    const element = svgRef.current;
+    if (element === null) {
+      return () => {};
+    }
+
+    element.addEventListener('mouseout', handleMouseOut);
+    element.addEventListener('mousemove', handleMouseMove);
+    return () => {
+      element.removeEventListener('mouseout', handleMouseOut);
+      element.removeEventListener('mousemove', handleMouseMove);
+    };
+  }, [handleMouseMove, handleMouseOut, svgRef]);
+
+  return mousePosition;
+};
+
+export type TooltipProps = {
+  /**
+   * Select the kinf of tooltip to display
+   * - 'item': Shows data about the item below the mouse.
+   * - 'axis': Shows vlaues associated to the hovered x value
+   * - 'none': Does nto display tooltip
+   * @default 'item'
+   */
+  trigger?: 'item' | 'axis' | 'none';
```

Comment:
Would anyone ever want having multiple tooltip triggers? 🤔

Reply:
I don't think so. You could cumulate the item interaction (highlighting a series) with an axis tooltip (highlighting values for a given axis) but having multiple trigger would mean multiple tooltip

The trigger API comes from Echarts

Reply:
Gotcha, thanks for the reply. 👌